internal/fsapi/file.go

```diff
@@ -380,3 +375,59 @@ type File interface {
 	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/close.html
 	Close() syscall.Errno
 }
+
+// Readdir is the status of a prior fs.ReadDirFile call.
+//
+// TODO:
+//   - Verify if it is possible to reduce the need to reopen the whole file.
+//   - Verify if it is useful to expose a Close() method.
```

Comment:
This feels like you should do it as adding io.Closer later is dodgy. It seems unlikely we'll not end in a position of needing to close an underlying fd, right?

Reply:
the thing is, the underlying FD is owned by the file you open, it is not created contextually to the Readdir instance. i.e., you do 

```
f, _ := sysfs.OpenFile()
rd := f.Readdir()
```

so the `Close()` method would only serve the purpose of "closing" the Readdir instance, which at this time I feel is kinda redundant. We can still add a `Close()` method that currently won't do anything in case we need it in further refactoring maybe?

Reply:
I have experimented today with actually cloning the File instance https://github.com/tetratelabs/wazero/pull/1529#issuecomment-1603951828 and I ended up again in the Windows/go1.18 world of pain. So I think for the time being I am going to add and document the Close method, but I am going to refrain from actually adding a meaningful implementation (will just return 0 everywhere), but I will do a follow-up PR with a proper implementation

the main issue is that if we do clone the FD, it is extremely important that the file is closed properly, otherwise the tests on Windows will often fail with a "TempDir RemoveAll cleanup: remove C:\Some\Dir: The process cannot access the file because it is being used by another process."

Reply:
ps remember Go 1.18 will fall off our support list in a couple months when go 1.21 is out!

Reply:
yeah that thought kept me sane 🤣